### PR TITLE
Upgrade to 1.0.3 and total rewriting

### PR DESCRIPTION
**Pull request by Jibec**
I can't understand why Upgrade and Backup/Restore doesn't work yet, but I'll fix that.
If anyone has suggestion!

## Problem
- *Description of why you made this PR*

## Solution
- *And how you fix that*

## PR Status
*Obviously, you should really check these affirmations*  
Work finished. Package_check, basic tests and upgrade from last version OK.  
Could be reviewed and tested.

## Validation
---
*Minor decision*
- [x] **Upgrade previous version** : Maniack C
- [x] **Code review** : frju365 
- [x] **Approval (LGTM)** : Maniack C
- [x] **Approval (LGTM)** : frju365
- [x] **CI succeeded** : Maniack C
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

**Comment by Jibec** [comment]
This now fully works, upgrade from existing app needs to be tested!

**Comment by Jibec** [comment]
Ok, this package is ready for review. I finally found a solution for mails.

**Comment by maniackcrudelis** [comment]
@YunoHost-Apps/apps-group

File: manifest.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
     "name": "OpenSondage",
     "id": "opensondage",
     "packaging_format": 1,
+    "url": "",
```
Comment (maniackcrudelis) [inline manifest.json]: https://git.framasoft.org/framasoft/framadate could be better than nothing.

File: scripts/install [hunk 2]
```diff
@@ -1,98 +1,182 @@
...
-sudo yunohost app addaccess $app -u $admin
-
-# Modify Nginx configuration file and copy it to Nginx conf directory
-sed -i "s@PATHTOCHANGE@$path@g" ../conf/nginx.conf
-sed -i "s@ALIASTOCHANGE@$final_path/@g" ../conf/nginx.conf
-sed -i "s@YNH_APP_INSTANCE_NAME@$app@g" ../conf/nginx.conf
-sudo cp ../conf/nginx.conf /etc/nginx/conf.d/$domain.d/$app.conf
-# Reload Nginx and regenerate SSOwat conf
-sudo service nginx reload
+chown -R "$app": "$final_path"
```
Comment (maniackcrudelis) [inline scripts/install]: Can you restrain the permission of $app only at the strict minimum

File: scripts/upgrade [hunk 3]
```diff
@@ -1,103 +1,203 @@
...
+#=================================================
+
+cp ../img/logo.png "$final_path/images"
+
+#=================================================
+# STORE THE CHECKSUM OF THE CONFIG FILE
+#=================================================
+
+# Verify the checksum and backup the file if it's different
+ynh_backup_if_checksum_is_different "$config"
```
Comment (maniackcrudelis) [inline scripts/upgrade]: That's too late, you should verify the checksum before the line 134. Here the config file is already overwritten.

**Comment by Jibec** [comment]
Everything should be fixed :)

**Comment by maniackcrudelis** [comment]
Package check OK, upgrade OK.
But... there's an error upstream with opensondage.
```
Attention : PHP Notice:  Undefined index: SERVER_PORT in /var/www/opensondage/app/classes/Framadate/Utils.php on line 27
Attention : PHP Notice:  Undefined index: SERVER_NAME in /var/www/opensondage/app/inc/smarty.php on line 32
```
Seems to work anyway.

**Comment by anmol26s** [comment]
Working :+1: 
Thanks

**Pull request merged**